vnext/Microsoft.ReactNative/Fabric/Composition/Composition.Input.h

```diff
@@ -28,4 +33,123 @@ struct KeyRoutedEventArgs : winrt::implements<
+  bool IsInverted() noexcept;
+  bool IsLeftButtonPressed() noexcept;
+  bool IsMiddleButtonPressed() noexcept;
+  bool IsPrimary() noexcept;
+  bool IsRightButtonPressed() noexcept;
+  bool IsXButton1Pressed() noexcept;
+  bool IsXButton2Pressed() noexcept;
+  int32_t MouseWheelDelta() noexcept;
+  float Orientation() noexcept;
+  PointerUpdateKind PointerUpdateKind() noexcept;
+  float Pressure() noexcept;
+  bool TouchConfidence() noexcept;
+  float Twist() noexcept;
+  float XTilt() noexcept;
+  float YTilt() noexcept;
+
+ private:
+  winrt::Windows::UI::Input::PointerPointProperties m_systemPPP{nullptr};
+
+  // When not using m_systemPPP
+  bool m_isBarrelButtonPressed : 1;
+  bool m_isCanceled : 1;
+  bool m_isEraser : 1;
+  bool m_isHorizontalMouseWheel : 1;
+  bool m_isInRange : 1;
+  bool m_isInverted : 1;
+  bool m_isLeftButtonPressed : 1;
+  bool m_isMiddleButtonPressed : 1;
+  bool m_isPrimary : 1;
+  bool m_isRightButtonPressed : 1;
+  bool m_isXButton1Pressed : 1;
+  bool m_isXButton2Pressed : 1;
+  bool m_touchConfidence : 1;
+  int32_t m_mouseWheelDelta;
+  float m_orientation;
+  winrt::Microsoft::ReactNative::Composition::Input::PointerUpdateKind m_pointerUpdateKind;
+  float m_pressure;
+  float m_twist;
+  float m_xTilt;
+  float m_yTilt;
+};
+
+struct PointerPoint : PointerPointT<PointerPoint> {
+  PointerPoint(const winrt::Windows::UI::Input::PointerPoint &pp);
+  PointerPoint(uint32_t msg, uint64_t wParam, int64_t lParam);
+
+  uint32_t FrameId() noexcept;
+  bool IsInContact() noexcept;
+  PointerDeviceType PointerDeviceType() noexcept;
+  uint32_t PointerId() noexcept;
+  winrt::Windows::Foundation::Point Position() noexcept;
+  winrt::Microsoft::ReactNative::Composition::Input::PointerPointProperties Properties() noexcept;
+  uint64_t Timestamp() noexcept;
+  winrt::Microsoft::ReactNative::Composition::Input::PointerPoint GetTransformedPoint(
+      const IPointerPointTransform &transform) noexcept;
+
+ private:
+  bool IsPointerMessage(uint32_t message) noexcept;
+
+  // Windows::Input
+  winrt::Windows::UI::Input::PointerPoint m_systemPP{nullptr};
+  // WM_POINTER*
+  POINTER_INFO m_pi = {0};
+  // WM_*MOUSE
+  uint32_t m_msg;
+  uint64_t m_wParam;
+  int64_t m_lParam;
+};
+
+struct PointerRoutedEventArgs : PointerRoutedEventArgsT<PointerRoutedEventArgs> {
+  PointerRoutedEventArgs(
+      facebook::react::Tag tag,
+      const winrt::Microsoft::ReactNative::Composition::Input::PointerPoint &pp,
+      const winrt::Windows::System::VirtualKeyModifiers &virtualKeyModifiers);
+
+  int32_t OriginalSource() noexcept;
+  winrt::Microsoft::ReactNative::Composition::Input::PointerPoint GetCurrentPoint(int32_t tag) noexcept;
+  bool Handled() noexcept;
+  void Handled(bool value) noexcept;
+  winrt::Windows::System::VirtualKeyModifiers KeyModifiers() noexcept;
+
+ private:
+  facebook::react::Tag m_tag{-1};
+  bool m_handled{false};
+  winrt::Microsoft::ReactNative::Composition::Input::PointerPoint m_pp{nullptr};
```

Comment:
nit: maybe rename to something less abbreviated?

```suggestion
  winrt::Microsoft::ReactNative::Composition::Input::PointerPoint m_pointerPoint{nullptr};
```